src/compiler/transformers/generators.ts

```diff
@@ -2528,7 +2528,7 @@ export function transformGenerators(context: TransformationContext): (x: SourceF
                 labelExpressions = [];
             }
 
-            const expression = factory.createNumericLiteral(-1);
+            const expression = factory.createNumericLiteral(Infinity);

```

Comment:
This shows how `createNumericLiteral` is still a somewhat weird thing. `Infinity` is allowed because parsed numbers are "normalized" to text here:
https://github.com/microsoft/TypeScript/blob/83f3abda153f0f567e296a3c75cdca37c7eb065d/src/compiler/scanner.ts#L1765-L1773

So a sufficiently long number in the input can be stringified to `Infinity` 🤷‍♂️ I find it weird an there is [an issue](https://github.com/microsoft/TypeScript/issues/33298) about it that is labeled as a bug. While it doesn't mention `Infinity` case the issue there is basically related to the same "normalization". 

This issue is certainly fixable but I'm not 100% sure if you'd really want to fix it.

Reply:
Don't change this to `Infinity` since we really should consider that invalid as well. I'd just use a large value that we're unlikely to actually hit in a generator, something like `2 ** 32 - 1` (max uint32). The only reason it's set to `-1` is to make it obvious something went wrong in tests.

Reply:
> Don't change this to Infinity

Ye, I committed it this way specifically to add this PR comment 😅 

> since we really should consider that invalid as well.

Strong agree, I had versions of this work that forbid this as well but decided to roll them back and only start the conversation about this for now. I'm open to following up with further changes but it seems that the mentioned "normalization" of parsed numeric literals is relied upon in a couple of places. They can be adjusted but I'm not sure if this is something you'd like to do. In a way, it would be easiest to just drop this whole normalization altogether and just cast parsed text to string in those other places (to produce correct resolved property names etc).

> I'd just use a large value that we're unlikely to actually hit in a generator, something like 2 ** 32 - 1 (max uint32).

pushed out a change with `Number.MAX_SAFE_INTEGER`